File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EBackgroundLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EBackgroundLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl1
            AppTheme.STANDARD -> R.color.cstl1
            AppTheme.BLUE -> R.color.cbl1
            AppTheme.GREEN -> R.color.cgl1
            AppTheme.RED -> R.color.crl1
            AppTheme.NOTHING -> R.color.cnol1
        }
    }
}

File: lib/layer2_core/light/ebackground_light.dart
import 'package:flutter/material.dart';

// Import RAW Light Values
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

import '../../layer3_logic/design_engine_controller.dart'; // Für AppTheme enum

/// Layer 2: Core Tokens - eBackground (Light Mode)
/// Mappt das gewählte Theme auf die entsprechende Konstante aus Schicht 1.
class EBackgroundLight {
  static Color getColor(AppTheme activeTheme) {
    switch (activeTheme) {
      case AppTheme.system:
        return csyl1;
      case AppTheme.standard:
        return cstl1;
      case AppTheme.blue:
        return cbl1;
      case AppTheme.green:
        return cgl1;
      case AppTheme.red:
        return crl1;
      case AppTheme.nothing:
        return cnol1;
    }
  }
}
